python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -517,3 +495,45 @@ def _private_display(self, reg_slice, this_format, default_width, markup):
             ds += markup(this_format % tuple(vals), colour)
 
         return ds
+
+    @staticmethod
+    def JobHasChildrenTest(jobDirectory):
+        """ Return True/False if given (job?) object has children associated with it
+        Args:
+            jobDirectory (str): name of folder to be examined
+        """
+
+        count = SubJobXMLList.countSubJobDirs(jobDirectory)
+        if count:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def countSubJobDirs(jobDirectory):
+        """ I'm a function which returns a number, my number corresponds to the amount of sequentially listed numerically named folders exiting within 'jobDirectory'
+        Args:
+            jobDirectory (str): name of folder to be examined
+        """
+
+        jobDirectoryList = listdir( jobDirectory )
+
+        subjob_count=0
+        while True:
+            if str(subjob_count) in jobDirectoryList:
+                expected_folder = path.join(jobDirectory, str(subjob_count))
+                if path.isdir(expected_folder):
+                    subjob_count=subjob_count+1
+                    continue
+                else:
+                    break
+            else:
+                break
+
+            #subjob_data = self.__get_dataFile(str(i))
+            #if path.isfile(subjob_data)):
+            #    subjob_count = subjob_count + 1
+            #i += 1
+
+        return subjob_count
```

Comment:
For starters, [`os.listdir` is not guaranteed to be ordered](https://docs.python.org/2.7/library/os.html?highlight=listdir#os.listdir).

Secondly, does this really need to only be sequentially numbered directories? If not, this could just be:
```python
@staticmethod
def countSubJobDirs(jobDirectory):
    return len(dir for dir in listdir(jobDirectory) if dir.isdigit())
    
```